e2etest/declarativeHelpers.go

```diff
@@ -481,6 +493,9 @@ type hookHelper interface {
 	// CreateFiles creates the specified files (overwriting any that are already there of the same name)
 	CreateFiles(fs testFiles, atSource bool)
 
+	// CreateFile creates a specified file (overwriting what was already there of the same name)
+	CreateFile(f *testObject, atSource bool)
```

Comment:
I see that `CreateFile()` is creating file/folder in the middle of transfer. Can you please write proper comment to explain the behaviour here as it may create confusion with `CreateFiles()`. 

Another thing: Can we not refactor `CreateFiles()` to adjust the required creation of files/folders? 
I think adding one more interface in `testFiles` could have made the design clearer.
```go
type testFiles struct {
     shouldCreateWhileTrasfer []interface{}
}
```

Right now we're redundantly rewriting the exact function for all the resource types. i.e. Local, Azure File, Blob, Dummy.

Reply:
CreateFile reuses the same underlying code as CreateFiles. I don't think there's really any code duplication to fix here.

Regarding shouldCreateDuringTransfer-- I think that's reinventing the wheel and creating something less useful at that. The hooks struct is already intended for this kinda stuff.